app/uk/gov/hmrc/fileupload/controllers/AuthBasicModule.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.fileupload.controllers
+
+import com.google.common.base.Charsets
+import com.google.common.io.BaseEncoding
+import play.api.http.HeaderNames
+import play.api.mvc.Results.Status
+import play.api.mvc.{RequestHeader, Result}
+
+import scala.concurrent.Future
+
+trait AuthBasicModule {
```

Comment:
the usual name for this is BasicAuth rather than AuthBasic however I'm being picky here